### PR TITLE
Update segment.py

### PR DESCRIPTION
**Pull request by objorkman**
* Change `async` to `non_blocking` for Python 3.7+
* Add dim=0 for softmax (https://discuss.pytorch.org/t/implicit-dimension-choice-for-softmax-warning/12314/10)

### CODE REVIEW

**Review by ICGog — APPROVED** [review]
Thank you for the changes! LGTM!
